File: demo-web-java/src/main/java/com/example/demo/repository/UserRepository.java
package com.example.demowebjava.repository;

import com.example.demowebjava.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long>, UserRepositoryCustom {

  User findByUsername(String username);

}

File: demo-web-scala/src/main/scala/com/example/demo/repository/UserRepository.scala
package com.example.demo.repository

import com.example.demo.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository


@Repository
trait UserRepository extends JpaRepository[User, java.lang.Long] {
  def findByUsername(username: String): User
}
